bigquery/google/cloud/bigquery/job.py

```diff
@@ -2009,16 +2018,28 @@ def default_dataset(self):
 
     @default_dataset.setter
     def default_dataset(self, value):
-        resource = None
-        if value is not None:
-            resource = value.to_api_repr()
+        if value is None:
+            self._set_sub_prop("defaultDataset", None)
+            return
+
+        if isinstance(value, six.string_types):
+            value = DatasetReference.from_string(value)
+
+        resource = value.to_api_repr()
         self._set_sub_prop("defaultDataset", resource)
 
     @property
     def destination(self):
         """google.cloud.bigquery.table.TableReference: table where results are
         written or :data:`None` if not set.
 
+        The ``destination`` setter accepts:
+
+        - a :class:`~google.cloud.bigquery.table.TableReference`, or
+        - a :class:`str` of the fully-qualified table ID in standard SQL
+          format. The value must included a project ID, dataset ID, and table
+          ID, each separated by ``.``.
```

Comment:
Same comment as above - add example

Reply:
Done in https://github.com/googleapis/google-cloud-python/pull/7560/commits/3ab95d4ad7da8b2be27171716e294f024882536d.